File: rename.py
import os
import re

directory = r"e:\web\flurbix"

def process_content(content):
    # Step 1: Extract CDN URLs and replace with placeholders
    cdn_pattern = re.compile(r'https://[^"\'\s)]*website-files\.com/[^"\'\s)<>]+')
    placeholders = {}
    
    def repl_cdn(match):
        idx = len(placeholders)
        key = f"__CDN_URL_{idx}__"
        placeholders[key] = match.group(0)
        return key

    content = cdn_pattern.sub(repl_cdn, content)
    
    # Step 2: Replace brand names
    content = content.replace("Auxia", "Flurbix")
    content = content.replace("AUXIA", "FLURBIX")
    content = content.replace("auxia", "flurbix")
    
    # Step 3: Restore CDN URLs
    for key, val in placeholders.items():
        content = content.replace(key, val)
        
    return content

count = 0
for root, dirs, files in os.walk(directory):
    # skip .git or other hidden dirs if any
    if ".git" in root or ".gemini" in root:
        continue
    for file in files:
        if file.endswith(".html") or file.endswith(".txt"):
            filepath = os.path.join(root, file)
            with open(filepath, "r", encoding="utf-8") as f:
                try:
                    content = f.read()
                except UnicodeDecodeError:
                    continue
            
            new_content = process_content(content)
            
            if new_content != content:
                with open(filepath, "w", encoding="utf-8") as f:
                    f.write(new_content)
                print(f"Updated: {filepath}")
                count += 1

print(f"Done. Updated {count} files.")

File: rename.js
const fs = require('fs');
const path = require('path');

const directory = 'e:\\web\\flurbix';

function processContent(content) {
    const cdnPattern = /https:\/\/[^"'\s)]*website-files\.com\/[^"'\s)<>]+/g;
    const placeholders = {};
    let idx = 0;
    
    // Step 1: Replace CDN URLs
    content = content.replace(cdnPattern, (match) => {
        const key = `__CDN_URL_${idx}__`;
        placeholders[key] = match;
        idx++;
        return key;
    });
    
    // Step 2: Replace brand names
    content = content.replace(/Auxia/g, 'Flurbix');
    content = content.replace(/AUXIA/g, 'FLURBIX');
    content = content.replace(/auxia/g, 'flurbix');
    
    // Step 3: Restore CDN URLs
    for (const [key, val] of Object.entries(placeholders)) {
        content = content.replace(key, val);
    }
    
    return content;
}

function walk(dir) {
    let count = 0;
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filepath = path.join(dir, file);
        const stat = fs.statSync(filepath);
        if (stat.isDirectory()) {
            if (file === '.git' || file === '.gemini' || file === 'node_modules') continue;
            count += walk(filepath);
        } else {
            if (file.endsWith('.html') || file.endsWith('.txt')) {
                const content = fs.readFileSync(filepath, 'utf8');
                const newContent = processContent(content);
                if (newContent !== content) {
                    fs.writeFileSync(filepath, newContent, 'utf8');
                    console.log(`Updated: ${filepath}`);
                    count++;
                }
            }
        }
    }
    return count;
}

const count = walk(directory);
console.log(`Done. Updated ${count} files.`);
